feat: add zero-conf toggle

kiocos opened the pull request:
Closes #346 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Zero-Conf toggle in Security settings to enable automatic claiming of eligible swaps.
  * Claiming API added to the payment context to coordinate cooperative/server-assisted claims.

* **Localization**
  * Added Zero-Conf label and tooltip translations; removed many legacy swap-related keys.

* **Behavior Changes**
  * Rescue list and claim eligibility respect Zero-Conf state.

* **Refactor**
  * Claim logic moved from UI checker into the payment/context layer.

* **Tests**
  * Unit and E2E tests updated and new E2E scenarios added for Zero-Conf auto-claim.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

michael1011 commented on src/utils/rescue.ts:
```diff
@@ -95,7 +97,7 @@ export const isSwapClaimable = ({
             const statuses = [
                 swapStatusPending.TransactionConfirmed,
-                swapStatusPending.TransactionMempool,
+                ...(zeroConf ? [swapStatusPending.TransactionMempool] : []),
```
Why not just push to it?

michael1011 commented on src/utils/rescue.ts:
```diff
@@ -107,7 +109,9 @@ export const isSwapClaimable = ({
-                swapStatusPending.TransactionServerMempool,
+                ...(zeroConf
+                    ? [swapStatusPending.TransactionServerMempool]
+                    : []),
```
Same here

kiocos commented:
@michael1011 in order to make the claim functionality reactive to the zeroConf toggle, I moved `claimSwap` from `SwapChecker` to `context/Pay.tsx` to share the claiming logic.

I've added `const claimingSwaps = new Set<string>();` because claiming is async and the WebApp receives a WebSocket message regarding the swap status update, which triggers another claim.

The duplicated claim request was being performed even before the code below (from the first claim) finished executing:
```
claimedSwap.claimTx = res.claimTx;
await setSwapStorage(claimedSwap);
```
Having the `claimTx` set would have prevented a duplicated request, if it was available earlier.
The `claimingSwaps` was the solution I found to fix this.

michael1011 commented on src/pages/RescueExternal.tsx:
```diff
@@ -148,7 +148,7 @@ export const RefundBtcLike = () => {
         async (swaps: SomeSwap[]) => {
             setLoading(true);
-            return await createRescueList(swaps).finally(() =>
+            return await createRescueList(swaps, zeroConf()).finally(() =>
```
I would also also accept 0-conf in the external rescues

michael1011 commented on src/context/Pay.tsx:
```diff
@@ -45,6 +100,198 @@ const PayProvider = (props: { children: JSX.Element }) => {
+            currentSwap.type === SwapType.Chain &&
+            (data.status === swapStatusSuccess.TransactionClaimed ||
+                (zeroConf() &&
+                    data.status === swapStatusPending.TransactionServerMempool))
```
Same here

kiocos replied:
If I turn zero-conf on in this situation, it wouldn't autoclaim with the previous condition because its status is `TransactionServerMempool` and not `TransactionClaimed`. The same for the other thread.
<img width="506" height="363" alt="image" src="https://github.com/user-attachments/assets/e80bc8df-205c-48bc-bf82-9ae812138b02" />

michael1011 replied:
Comment in the code please
